test/auth.test.js

```diff
@@ -0,0 +1,87 @@
+import assert from 'assert'
+import jsdom from 'jsdom-global';
+
+// initialize some constants
+const url = 'http://localhost/';
+const user = {name: "John Doe"};
+const payload = Buffer.from(JSON.stringify(user)).toString('base64').replace(/=/g, '');
+const jwt = `header.${payload}.signature`;
+const token = `#access_token=${jwt}`;
+const type = '&token=Bearer';
+const state = `&state=test`;
+
+// initilize the window, document and localStorage objects
+jsdom('', {url: url + token + type + state});
+global.localStorage = window.localStorage;
+
+// load the component
+import Auth from '../src/auth.js';
+
+describe('Auth', function () {
+  
+  describe('state', function () {
+    it('should initialize the state', function () {
+      let state = Auth.state();
+      assert.ok(state.length > 0);
+      assert.equal(Auth.state(), state);
+      Auth.state('test');
+      assert.equal(Auth.state(), 'test');
+    });
+  });
+
+  describe('getUser, setUser and removeUser', function () {
+    it('should get, set and remove the user', function () {
+      Auth.removeUser();
+      assert.deepEqual(Auth.getUser(), null);
+      Auth.setUser(user);
+      assert.deepEqual(Auth.getUser(), user);
+      Auth.removeUser();
+      assert.deepEqual(Auth.getUser(), null);
+    });
+  });
+
+  describe('authenticate', function () {
+    it('should wait until the user authenticates', async function () {
+      Auth.removeUser()
+      setInterval(() => Auth.setUser(user), 100);
```

Comment:
Let's use something a bit bigger to test the `while` loop.
```suggestion
      setInterval(() => Auth.setUser(user), 120);
```